File: grazi/src/main/java/tanvd.grazi/GraziToolConfigurable.java
package tanvd.grazi;

import com.intellij.openapi.options.*;
import com.intellij.openapi.project.*;
import org.jetbrains.annotations.*;

public class GraziToolConfigurable extends ConfigurableBase<GraziSettingsPanel, GraziApplicationSettings> {
    public GraziToolConfigurable() {
        super("reference.settingsdialog.project.grazi", "Grazi", null);
    }

    @NotNull
    @Override
    protected GraziApplicationSettings getSettings() {
        return GraziApplicationSettings.getInstance();
    }

    @Override
    protected GraziSettingsPanel createUi() {
        return new GraziSettingsPanel();
    }
}

File: grazi/src/main/kotlin/tanvd/grazi/ui/GraziToolConfigurable.kt
package tanvd.grazi.ui

import com.intellij.openapi.options.ConfigurableBase

class GraziToolConfigurable : ConfigurableBase<GraziSettingsPanel, GraziApplicationSettings>("reference.settingsdialog.project.grazi", "Grazi", null) {

    override fun getSettings(): GraziApplicationSettings {
        return GraziApplicationSettings.instance
    }

    override fun createUi(): GraziSettingsPanel {
        return GraziSettingsPanel()
    }
}
